src/android/com/totalpave/cordova/insets/Insets.java

```diff
@@ -45,11 +46,59 @@ protected void pluginInitialize() {
                     // Ideally, we'd import this, but it shares the same name as our plugin
                     int insetTypes = WindowInsetsCompat.Type.displayCutout() | WindowInsetsCompat.Type.systemBars();
                     androidx.core.graphics.Insets insets = insetProvider.getInsets(insetTypes);
+
+                    double topLeftRadius = 0.0;
+                    double topRightRadius = 0.0;
+                    double botLeftRadius = 0.0;
+                    double botRightRadius = 0.0;
+
+                    WindowInsets sourceInsets = insetProvider.toWindowInsets();
+                    if (sourceInsets != null) {
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                            RoundedCorner topLeft = sourceInsets.getRoundedCorner(RoundedCorner.POSITION_TOP_LEFT);
+                            RoundedCorner topRight = sourceInsets.getRoundedCorner(RoundedCorner.POSITION_TOP_RIGHT);
+                            RoundedCorner botLeft = sourceInsets.getRoundedCorner(RoundedCorner.POSITION_BOTTOM_LEFT);
+                            RoundedCorner botRight = sourceInsets.getRoundedCorner(RoundedCorner.POSITION_BOTTOM_RIGHT);
+
+                            if (topLeft != null) {
+                                int radius = topLeft.getRadius();
+                                topLeftRadius = (double)radius / density;
+                            }
+
+                            if (topRight != null) {
+                                int radius = topRight.getRadius();
+                                topRightRadius = (double)radius / density;
+                            }
+
+                            if (botLeft != null) {
+                                int radius = botLeft.getRadius();
+                                botLeftRadius = (double)radius / density;
+                            }
+
+                            if (botRight != null) {
+                                int radius = botRight.getRadius();
+                                botRightRadius = (double)radius / density;
+                            }
+                        }
+                    }
+
+                    double top = insets.top / density;
+                    double right = insets.right / density;
+                    double bottom = insets.bottom / density;
+                    double left = insets.left / density;
+
+                    // these insets kind of already account for rounded corners by their padded
+                    // space. Therefore, we will take the maximum value out of the two.
```

Comment:
If insets already account for rounded corners why is this extra work required? Include explanation in the comment itself.